File: backend/src/main/java/backend/service/UserService.java
package backend.service;

import backend.dto.request.UserCreationRequest;
import backend.dto.request.UserUpdateRequest;
import backend.dto.response.UserResponse;

import java.util.List;

public interface UserService {
    UserResponse createUser(UserCreationRequest request);
    UserResponse updateUser(Long id, UserUpdateRequest request);
    List<UserResponse> getAllUsers();
    UserResponse getUserById(Long id);
    void deleteUser(Long id);
    UserResponse getMyInfo();
}

File: frontend/src/services/userService.js
import apiClient from "./apiClient";

const userService = {
  // GET /users (all users — used for admin management & staff dropdowns)
  getAllUsers: async () => {
    const response = await apiClient.get("/users");
    return response.data.result;
  },

  // GET /users/:id
  getUserById: async (id) => {
    const response = await apiClient.get(`/users/${id}`);
    return response.data.result;
  },

  // POST /users — create user
  createUser: async (data) => {
    const response = await apiClient.post("/users", data);
    return response.data.result;
  },

  // PUT /users/:id — update user
  updateUser: async (id, data) => {
    const response = await apiClient.put(`/users/${id}`, data);
    return response.data.result;
  },

  // DELETE /users/:id
  deleteUser: async (id) => {
    await apiClient.delete(`/users/${id}`);
  },

  // POST /api/auth/change-password
  changePassword: async (data) => {
    await apiClient.post("/api/auth/change-password", data);
  },
};

export default userService;
